### ISSUE TITLE
JSII: node process consumes a lot of CPU

### ISSUE THREAD

**Issue by metametadata** [aws/aws-cdk#12069]
`node` process recently started consuming a lot of CPU even when there's no CDK code executed. Reproduced in Clojure.

### Reproduction Steps

1) Install `clj`, e.g. via `brew install clojure/tools/clojure`.
1) Execute simple CDK code:

      > clj -Sdeps '{:deps {software.amazon.awscdk/core {:mvn/version "1.61.1"}}}' -e '(do (import software.amazon.awscdk.core.App) (App.) (println "Press Enter to exit...") (.read System/in) nil)'

1) Wait for `Press Enter to exit...` output.
1) Run `htop`, Activity Monitor or any similar app.

### What did you expect to happen?

`node` process for `jsii-runtime.js` takes less than 1% of CPU.

### What actually happened?

~99% of CPU used. Example from `htop`:

```
PID USER      PRI  NI  VIRT   RES S CPU% MEM%   TIME+  Command
35650 yuri       17   0 4576M 64776 ? 99.4  0.4  4:24.99 node /var/folders/6g/9k9wc34n0hv_cn_1sq78xfdc0000gn/T/jsii-java-runtime15336424386656685432/jsii-runtime.js
```

### Environment

  - **CDK CLI Version  :** 1.77.0
  - **Framework Version:** 1.61.1 (**`jsii` 1.16.0**)
  - **Node.js Version:** v15.4.0
  - **OS               :** macOS 10.15.6
  - **Language (Version):** Clojure 1.10.1, Java 11.0.9

### Other

Is not reproducible in `jsii` `1.15.0` and thus could be related to https://github.com/aws/jsii/issues/2284. Example:

> clj -Sdeps '{:deps {software.amazon.awscdk/core {:mvn/version "1.61.1" :exclusions [software.amazon.jsii/jsii-runtime]} software.amazon.jsii/jsii-runtime {:mvn/version "1.15.0"}}}' -e '(do (import software.amazon.awscdk.core.App) (App.) (println "Press Enter to exit...") (.read System/in) nil)'

`htop` output for `node`:

```
  PID USER      PRI  NI  VIRT   RES S CPU% MEM%   TIME+  Command 
38017 yuri       17   0 4518M 46716 ?  0.4  0.3  0:00.64 node /var/folders/6g/9k9wc34n0hv_cn_1sq78xfdc0000gn/T/jsii-java-runtime14064155404264507295/jsii-runtime.js
```

--- 

This is :bug: Bug Report

**Comment by NetaNir** [comment]
Yes, we believe the root cause of this is the added `sleep` in the runtime code, it was removed and the fix was released in jsii version `1.15`.  Please let us know if this persists  after upgrading to the latest jsii version.

**Comment by metametadata** [comment]
Based to my experiments, JSII 1.15.0 works and 1.16.0 (latest) is buggy.

**Comment by NetaNir** [comment]
Thanks, we will check it out

**Comment by RomainMuller** [comment]
I think you had a good time with the version _with_ `sleep` (that might be _very_ slow), and CPU thrashing with the version _without_ sleep (since it'll just loop until data becomes available).

I guess we may want to prioritise fixing this particular problem... because it's bad.

**Comment by metametadata** [comment]
Thank you for the fix, @RomainMuller.
